com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs

```diff
@@ -198,6 +200,10 @@ bool StripUnusedFeatures(ShaderFeatures features, Shader shader, ShaderSnippetDa
                 !IsFeatureEnabled(features, ShaderFeatures.LightLayers))
                 return true;
 
+            if (compilerData.shaderKeywordSet.IsEnabled(m_RenderPassEnabled) &&
+                !IsFeatureEnabled(features, ShaderFeatures.RenderPassEnabled))
+                return true;
```

Comment:
Some things to consider. 
 1) Not all platforms support this keyword. For platforms that don't we shouldn't add more variants.
 2) Is there value for this keyword to be changed at runtime? We have plan to rediscuss shader variants stripping and this one seems a good candidate to always be stripped either the enabled or disabled variant. Essentially this will avoid adding more variants to build. cc @pbbastian 
 3) What happens when a shader doesn't define the multi_compile for this?

Reply:
> Not all platforms support this keyword
you mean platforms that do not support RenderPass API natively? As it will still use the fb fetch macros to fallback

Regarding 2. - if we are playing it safe and want to have a proper fallback mechanism - runtime change might happen and would fallback from fb fetch to the old LOAD_TEXTURE2D_X calls. This is definitely not necessary, but i did think we want to have both paths available in case something wrong happens and we want to fallback from RenderPass path. 
I'll admit I'm not fully familiar with the shader stripping part in URP, but my goal was to always strip if RenderPass API option is not enabled. Let me know if i made a mistake in doing that.

Regarding 3. - if shader doesn't multicompile it (let's it uses our UniversalGBuffer pass), the object won't be written into the GBuffer4 (fake depth) and DeferredFog will be rendered over it during DeferredPass (or artifacts, but so far saw only deferred fog over it, but it's definitely undefined behaviour)